backend/functions/lib/adapters/gateways/dxtr/dxtr.gateway.helper.ts

```diff
@@ -30,3 +31,12 @@ export function parseTransactionDate(record: DxtrTransactionRecord): Date {
     parseInt(transactionDateDay),
   );
 }
+
+// Examples:
+// 1081201013220-10132            15JC
+// 1081231056523-10565            15IB00-0000000
+export function parseDebtorType(record: DxtrTransactionRecord): string {
+  const debtorTypeMatch = record.txRecord.match(/\d{13}-\d{5}\s+\d{2}\w{2}/);
```

Comment:
Why do we not just take the two characters from their place in the fixed width string as we did with the transaction dates previously?

Reply:
I just felt like this was a better way to grab the ID.  We can discuss.  Maybe it's because I didn't want to keep track of character columns.  I'm being lazy perhaps?

Reply:
Now that I've given it more thought, I believe my original use of regex was because, different types of transactions seem to have very different record formats.  If all we do is grab characters at a certain location, we are likely to be grabbing the wrong information.

Reply:
My main concern is with the relative strictness of the pattern. 13 digits, a dash, 5 digits, some amount of whitespace*, 2 digits, 2 word characters. With the exception of the whitespace, this is perhaps always correct for valid data. But is some of that whitespace placeholder? So is there any valid record in the real database that has some data in place of some of that whitespace?